shard-operator/src/main/java/com/redhat/service/smartevents/shard/operator/ObservabilitySetupService.java

```diff
@@ -0,0 +1,110 @@
+ * Changes to the underlying {@link ConfigProperty} values are handled by the CPaaS
+ * AddOn deployment process and {@link SecretRestartHandler}.
+ */
+@ApplicationScoped
+public class ObservabilitySetupService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ObservabilitySetupService.class);
+
+    protected static final String OBSERVABILITY_ACCESS_TOKEN = "access_token";
+    protected static final String OBSERVABILITY_REPOSITORY = "repository";
+    protected static final String OBSERVABILITY_CHANNEL = "channel";
+    protected static final String OBSERVABILITY_TAG = "tag";
+    protected static final String OBSERVABILITY_NAME = "smartevents-observability-configuration";
+    protected static final String OBSERVABILITY_NAMESPACE = "smartevents-observability";
+
+    private static final Base64.Encoder ENCODER = Base64.getEncoder();
+
+    @ConfigProperty(name = "event-bridge.observability.access_token")
+    protected Optional<String> optAccessToken;
+    String accessToken;
+
+    @ConfigProperty(name = "event-bridge.observability.repository")
+    protected Optional<String> optRepository;
+    String repository;
+
+    @ConfigProperty(name = "event-bridge.observability.channel")
+    protected Optional<String> optChannel;
+
+    @ConfigProperty(name = "event-bridge.observability.tag")
+    protected Optional<String> optTag;
+
+    @Inject
+    KubernetesClient kubernetesClient;
+
+    @Inject
+    TemplateProvider templateProvider;
+
+    void setupObservabilityComponent(@Observes StartupEvent event) {
+        createOrUpdateObservabilitySecret();
+    }
+
+    protected void createOrUpdateObservabilitySecret() {
+        extractConfigProperties();
+        if (!isObservabilityEnabled()) {
+            return;
+        }
+
+        Secret existing = kubernetesClient
+                .secrets()
+                .inNamespace(OBSERVABILITY_NAMESPACE)
+                .withName(OBSERVABILITY_NAME)
+                .get();
+
+        Secret expected = templateProvider.loadObservabilitySecretTemplate();
+        expected.getData().put(OBSERVABILITY_ACCESS_TOKEN,
+                ENCODER.encodeToString(accessToken.getBytes(StandardCharsets.UTF_8)));
+        expected.getData().put(OBSERVABILITY_REPOSITORY,
+                ENCODER.encodeToString(repository.getBytes(StandardCharsets.UTF_8)));
+        optChannel.ifPresent(channel -> expected.getData().put(OBSERVABILITY_CHANNEL,
+                ENCODER.encodeToString(channel.getBytes(StandardCharsets.UTF_8))));
+        optTag.ifPresent(tag -> expected.getData().put(OBSERVABILITY_TAG,
+                ENCODER.encodeToString(tag.getBytes(StandardCharsets.UTF_8))));
+
+        if (existing == null || !expected.getData().equals(existing.getData())) {
+            try {
+                kubernetesClient
+                        .secrets()
+                        .inNamespace(expected.getMetadata().getNamespace())
```

Comment:
@manstis Does this work if the `namespace` doesn't already exist?

Reply:
No. I now create the target `Namespace` if it does not exist and log the action.

The Observability Operator does not care in what namespace the Secret resides. Do we?

Reply:
No, I don't think we do to be honest as long as it's sensibly named by default.